### DISCUSSION TITLE
Ionic Capacitor Pre/Post Build Script

### DISCUSSION THREAD

**Discussion by jwburnside** [ionic-team/capacitor#3560] [Q&A] [answered]
In our Ionic Cordova platform we could run scripts with hooks such as after_prepare, after_plugin_add, etc. 

Is there any equivalent with Capacitor? For instance, I would like to upload a source map to bugsnag every time I do a 
**ionic capacitor run ios**

I suppose I could just add a command to run that script every time, but that seems tedious. 

Any feedback would be appreciated!

**Comment by imhoffd** [comment, marked as answer]
Capacitor doesn't have hooks, but the Ionic CLI [has a few hooks](https://ionicframework.com/docs/cli/configuration#hooks) that may be useful to you.

**Comment by jwburnside** [comment]
@dwieeb And one more followup question if you have a few minutes: Is there a way to allow the hook to be a typescript file as opposed to javascript file? We were using cordova-ts-config which no longer applies. Thanks!

**Reply by imhoffd** [reply]
The docs explain how to use JS or shell scripts for hooks. With shell scripts, you can use whatever you want.
